### DISCUSSION TITLE
Multiple virtualenvs

### DISCUSSION THREAD

**Discussion by pawamoy** [astral-sh/rye#438] [Q&A]
Is there a way to create and use one virtualenv per Python version?

Currently, if I change my toolchain pin and run `rye sync`, it recreates the venv from scratch. That means that if I want to run `pytest` on multiple Python versions, Rye will recreate the venv and reinstall all dependencies each time. I don't use tox or nox.

Ideally I'd like an option to tell Rye to suffix the venv name with `{major}{minor}`.

Example:

```bash
for py in 3.8 3.9 3.10 3.11; do
  rye pin --relaxed $py
  rye sync  # creates .venv38, .venv39, etc.
done
```

I could then run my tests suite on all supported Python versions:

```bash
for py in 3.8 3.9 3.10 3.11; do
  rye pin --relaxed $py
  rye run pytest tests
done
```

**Comment by doolio** [comment]
> Rye to suffix the venv name with {major}{minor}

+1 for this naming convention

For multiple environments [`hatch`](https://hatch.pypa.io/latest/environment/) does this well. It creates them by default but could be improved further to not include `pytest` in the environment where the package is installed.
